src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/ICodeModelService.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore
+{
+    public interface ICodeModelService
+    {
+        /// <summary>
+        /// Gets the metadata for a given model without using the Datacontext.
+        /// Without the datacontext, the metadata will consist of no Navigations/ Primary keys.
+        /// </summary>
+        /// <param name="modelType">Model type for the EF metadata has to be returned.</param>
+        /// <returns>Returns <see cref="ContextProcessingResult"/>.</returns>
+        Task<ContextProcessingResult> GetModelMetadata(ModelType modelType);
```

Comment:
It'd probably be a good idea to change all method names to follow the convention of using Async suffix, but currently the methods in the repo don't do that. Will create a separate PR for that.